city_scrapers/spiders/il_port_district.py

```diff
@@ -0,0 +1,183 @@
+
+                date = date.strip(" *")
+
+                title = meeting_types[i] if not special_meeting\
+                    else special_meeting + meeting_types[i]
+
+                classification = self._parse_classification(i, meeting_types[i])
+
+                start = self._parse_start(year, date, meeting_time)
+
+                agendas_links = self._parse_agendas_links(classification)
+
+                minutes_links = None
+
+                if classification == "Board":
+                    minutes_links = self._parse_minutes_links(start)
+                links = agendas_links if not minutes_links else\
+                    agendas_links + minutes_links
+
+                meeting = Meeting(
+                    title=title,
+                    description="",
+                    classification=classification,
+                    start=start,
+                    end=None,
+                    all_day=False,
+                    time_notes="",
+                    location=self.location,
+                    links=links,
+                    source=response.url
+                )
+
+                meeting["status"] = self._get_status(meeting)
+                meeting["id"] = self._get_id(meeting)
+
+                yield meeting
+
+    def parse_agendas(self, response):
+        file_names = response.xpath("//tr/td[@class='views-field views-field-title']/text()")\
+            .extract()
+        file_names = [x.strip("\n ") for x in file_names]
+        file_links = response.xpath("//tr/td/a[@class='file-download']/@href").extract()
+
+        self.files_list = []
+
+        for link, name in list(zip(file_links, file_names)):
+            self.files_list.append({'title': name, 'href': link})
+
+    def parse_minutes(self, response):
+        rows = response.xpath("//tr")
+        self.minutes_dict = {}
+        for row in rows:
+            file_name = row.xpath(".//td[@class='views-field views-field-title']/text()")\
+                .extract_first()
+            if not file_name:
+                continue
+
+            file_name = file_name.strip("\n ")
+            file_name_dt = re.findall(r"(?:.*?)(?:\d{4})", file_name)[0]
+            file_name_dt = datetime.datetime.strptime(file_name_dt, "%B %d, %Y")
+
+            file_link = row.xpath(".//td[@class='views-field views-field-field-file']/a/@href")\
+                .extract_first()
+
+            self.minutes_dict.setdefault(file_name_dt.date(), file_link)
+
+    def _parse_classification(self, i, meeting_types):
+        if "board" in meeting_types.lower():
+            return BOARD
+        elif "committee" in meeting_types.lower():
+            return COMMITTEE
+        else:
+            return NOT_CLASSIFIED
+
+    def _parse_start(self, year, date, meeting_time):
+        dt = " ".join([year, date, meeting_time])
+        return datetime.datetime.strptime(dt, "%Y %B %d %I:%M%p")
+
+    def _parse_agendas_links(self, classification):
```

Comment:
Looking at this more closely, it should only associate the agenda with the meeting that it corresponds to, but right now it's attaching the November agenda to every meeting. We should check the month here